API changes and improvements (#12)
psteinb/c2st: README.md

# c2st

## Two sample test using a ML classifier

Test whether sets of D-dimensional points are samples from the same
multivariate probability distribution.

The `c2st` function returns the accuracy of how well a binary classifier was
able to classify two sets of points `X` and `Y` while being trained on the
concatenated dataset `(X,Y)`. All samples of `X` have received the label `0`
and `Y` has received the label `1`.

A value close to 0.5 means that the classifier is not better than random
guessing, i.e. `X` and `Y` are likely from the same distribution. A value close
to 1 means the classifier was able to separate `X` and `Y`, so they are
probably samples from different distributions.

```py
>>> import numpy as np
>>> from c2st.check import c2st

>>> rng=np.random.default_rng(seed=123)

# same distribution (Gaussian N(0,1)), D=20, 1000 points each
>>> X=rng.normal(loc=0, scale=1, size=(1000,20))
>>> Y=rng.normal(loc=0, scale=1, size=(1000,20))
>>> c2st(X, Y)
0.4970122828225085

> # now shift the mean of Y a bit

Reviewer: ```suggestion
# now shift the mean of Y by 0.3
```

Reviewer: Let's be precise and help people not familiar with the `np.random` API to understand the example.

>>> Y=rng.normal(loc=0.3, scale=1, size=(1000,20))
0.6964673015530594

> # and some more

Reviewer: ```suggestion
# let's move Y more extensively away from X to 1.5
```

>>> Y=rng.normal(loc=1.5, scale=1, size=(1000,20))
>>> c2st(X, Y)
0.9994791666666666

> # or change the distribution's width

Reviewer: ```suggestion
# or change the distribution's variance, so we compare a narrow to a wide distribution at the same loc
```
